src/render.rs

```diff
@@ -523,15 +523,42 @@ impl<'a, ComponentId: Clone + Debug + Eq + Hash> Viewport<'a, ComponentId> {
         span_rect
     }
 
+    /// Draw a [`Line`] directly to the screen at `(x, y)` location.
+    /// If the contents would overflow the current mask,
+    /// then it is truncated with an ellipsis.
```

Comment:
I think this comment is no longer true and the argument to `draw_line` is not truncated with an ellipsis?

Reply:
Thanks for catching this!

After looking over this again, I noticed that `draw_text` returns `line_rect` whereas `draw_line` returns the `draw_rect` which will never be larger than the current mask. Should `draw_rect` instead return the rect of the `Line` it's rendering?

Reply:
[renamed `span_rect` to `line_rect`](https://github.com/arxanas/scm-record/compare/5fbf58be5a787cc8ffc046a073a3074536c2bc4e..e68fd2765442371b8c057fde5765bdf1ed7d0ed9) and [removed the old comment](https://github.com/arxanas/scm-record/compare/9a0bfe4d8fd92d088763d5f372c552dc3ed7c927..5fbf58be5a787cc8ffc046a073a3074536c2bc4e)

Reply:
> I noticed that `draw_text` returns `line_rect` whereas `draw_line` returns the `draw_rect` which will never be larger than the current mask. Should `draw_rect` instead return the rect of the `Line` it's rendering?

This behavior also required this [comparison change](https://github.com/arxanas/scm-record/pull/37/commits/bd9db4f5ed7271c3a4c0997d8c6d08bca35e5989#diff-878aa4086be78d3ed352164ac8f6d78eab13f2b72ca1c26a01016cde4f87fcc9R556), so potentially worth thinking about (maybe for a later PR? dunno if it's an issue here)

Reply:
Hmm, I left frustratingly little documentation for myself, but I think the intention was to return the logically-drawn area and not the physically drawn area, the idea being that you could take the result and immediately draw something adjacent to it. In that case, we would return the rect of the `Line` it's rendering rather than the masked rect.

Reply:
👍🏻 https://github.com/arxanas/scm-record/compare/e68fd2765442371b8c057fde5765bdf1ed7d0ed9..4f66b1ec1e39b57e4b91fb8a84f17b1d3b9255f5